pkg/function/backups_info.go

```diff
@@ -0,0 +1,166 @@
+	BackupsInfoEncryptionKeyArg = "encryptionKey"
+	// BackupsInfoPodOverrideArg contains pod specs to override default pod specs
+	BackupsInfoPodOverrideArg    = "podOverride"
+	BackupsInfoJobPrefix         = "get-backups-info-"
+	BackupsInfoFileCount         = "fileCount"
+	BackupsInfoSize              = "size"
+	BackupsInfoSnapshotIDs       = "snapshotIDs"
+	BackupsInfoPasswordIncorrect = "passwordIncorrect"
+	BackupsInfoRepoUnavailable   = "repoUnavailable"
+)
+
+func init() {
+	kanister.Register(&BackupsInfoFunc{})
+}
+
+var _ kanister.Func = (*BackupsInfoFunc)(nil)
+
+type BackupsInfoFunc struct{}
+
+func (*BackupsInfoFunc) Name() string {
+	return "BackupsInfo"
+}
+
+func BackupsInfo(ctx context.Context, cli kubernetes.Interface, tp param.TemplateParams, namespace, encryptionKey, targetPaths, jobPrefix string, podOverride sp.JSONMap) (map[string]interface{}, error) {
+	options := &kube.PodOptions{
+		Namespace:    namespace,
+		GenerateName: jobPrefix,
+		Image:        kanisterToolsImage,
+		Command:      []string{"sh", "-c", "tail -f /dev/null"},
+		PodOverride:  podOverride,
+	}
+	pr := kube.NewPodRunner(cli, options)
+	podFunc := BackupsInfoPodFunc(cli, tp, namespace, encryptionKey, targetPaths)
+	return pr.Run(ctx, podFunc)
+}
+
+func BackupsInfoPodFunc(cli kubernetes.Interface, tp param.TemplateParams, namespace, encryptionKey, targetPath string) func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
+	return func(ctx context.Context, pod *v1.Pod) (map[string]interface{}, error) {
+		// Wait for pod to reach running state
+		if err := kube.WaitForPodReady(ctx, cli, pod.Namespace, pod.Name); err != nil {
+			return nil, errors.Wrapf(err, "Failed while waiting for Pod %s to be ready", pod.Name)
+		}
+		pw, err := getPodWriter(cli, ctx, pod.Namespace, pod.Name, pod.Spec.Containers[0].Name, tp.Profile)
+		if err != nil {
+			return nil, err
+		}
+		defer cleanUpCredsFile(ctx, pw, pod.Namespace, pod.Name, pod.Spec.Containers[0].Name)
+		snapshotIDs, err := restic.GetSnapshotIDs(tp.Profile, cli, targetPath, encryptionKey, namespace, pod.Name, pod.Spec.Containers[0].Name)
+		if err != nil {
+			if strings.Contains(err.Error(), restic.PasswordIncorrect) {
+				return map[string]interface{}{
+						BackupsInfoSnapshotIDs:       nil,
+						BackupsInfoFileCount:         nil,
+						BackupsInfoSize:              nil,
+						BackupsInfoPasswordIncorrect: "true",
+						BackupsInfoRepoUnavailable:   nil,
+					},
+					nil
+			}
+			if strings.Contains(err.Error(), restic.RepoDoesNotExist) {
+				return map[string]interface{}{
+						BackupsInfoSnapshotIDs:       nil,
+						BackupsInfoFileCount:         nil,
+						BackupsInfoSize:              nil,
+						BackupsInfoPasswordIncorrect: nil,
+						BackupsInfoRepoUnavailable:   "true",
+					},
+					nil
+			}
+			return nil, err
+		}
+		cmd, err := restic.StatsCommandByID(tp.Profile, targetPath, "" /* get all snapshot stats */, DefaultStatsMode, encryptionKey)
```

Comment:
We're should get the repo size not the restore-size (`DefaultStatsMode` == `restore-size` )

Reply:
👍 Yep. Changing it to raw-data